### PR TITLE
HDDS-2846. Handle Datanode Pipeline & Container Reports reports in Recon.

### PR DESCRIPTION
**Pull request by avijayanhwx**
## What changes were proposed in this pull request?
Recon should be able to process Container and Pipeline reports from registered Datanodes. It still wont be able to understand new containers and pipelines though. Along with HDDS-2850, HDDS-2852 and HDDS-2869, Recon will be a fully functional passive SCM. 
In addition to the above, some refactoring work also has been done.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-2846

## How was this patch tested?
Manually tested on docker.
Created key (container) on ozone cluster, and verified Recon gets the container info in Datanode heartbeat.

### CODE REVIEW

**Comment by avijayanhwx** [comment]
@anuengineer  / @swagle Please review.

**Comment by avijayanhwx** [comment]
@nandakumar131 Please review.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconStorageContainerManagerFacade.java [hunk 1]
```diff
@@ -27,52 +28,99 @@
...
 import org.apache.hadoop.hdds.server.events.EventQueue;
+import org.apache.hadoop.io.IOUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Recon's 'lite' version of SCM.
  */
-public class ReconStorageContainerManager
+public class ReconStorageContainerManagerFacade
```
Comment (swagle) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconStorageContainerManagerFacade.java]: Minor nit: Should drop the Facade from the class name, a facade merely provides access to underlying subsystems minus the management functions which this class actually does in addition to abstracting out access.

Comment (avijayanhwx) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconStorageContainerManagerFacade.java]: Thanks for the review @swagle. I will address this in the next patch (HDDS-2869).

**Pull request merged**